tests/app/user/test_rest.py

```diff
@@ -1088,72 +1039,58 @@ def test_get_orgs_and_services_only_shows_users_orgs_and_services(admin_request,
     ]
 
 
-def test_find_users_by_email_finds_user_by_partial_email(notify_db, client):
+def test_find_users_by_email_finds_user_by_partial_email(notify_db_session, admin_request):
     create_user(email='findel.mestro@foo.com')
     create_user(email='me.ignorra@foo.com')
-    data = json.dumps({"email": "findel"})
-    auth_header = create_admin_authorization_header()
+    data = {"email": "findel"}
 
-    response = client.post(
-        url_for("user.find_users_by_email"),
-        data=data,
-        headers=[('Content-Type', 'application/json'), auth_header]
+    users = admin_request.post(
+        "user.find_users_by_email",
+        _data=data,
     )
-    users = json.loads(response.get_data(as_text=True))
 
-    assert response.status_code == 200
     assert len(users['data']) == 1
     assert users['data'][0]['email_address'] == 'findel.mestro@foo.com'
 
 
-def test_find_users_by_email_finds_user_by_full_email(notify_db, client):
+def test_find_users_by_email_finds_user_by_full_email(notify_db_session, admin_request):
     create_user(email='findel.mestro@foo.com')
     create_user(email='me.ignorra@foo.com')
-    data = json.dumps({"email": "findel.mestro@foo.com"})
-    auth_header = create_admin_authorization_header()
+    data = {"email": "findel.mestro@foo.com"}
 
-    response = client.post(
-        url_for("user.find_users_by_email"),
-        data=data,
-        headers=[('Content-Type', 'application/json'), auth_header]
+    users = admin_request.post(
+        "user.find_users_by_email",
+        _data=data,
     )
-    users = json.loads(response.get_data(as_text=True))
 
-    assert response.status_code == 200
     assert len(users['data']) == 1
     assert users['data'][0]['email_address'] == 'findel.mestro@foo.com'
 
 
-def test_find_users_by_email_handles_no_results(notify_db, client):
+def test_find_users_by_email_handles_no_results(notify_db_session, admin_request):
     create_user(email='findel.mestro@foo.com')
     create_user(email='me.ignorra@foo.com')
-    data = json.dumps({"email": "rogue"})
-    auth_header = create_admin_authorization_header()
+    data = {"email": "rogue"}
 
-    response = client.post(
-        url_for("user.find_users_by_email"),
-        data=data,
-        headers=[('Content-Type', 'application/json'), auth_header]
+    users = admin_request.post(
+        "user.find_users_by_email",
+        _data=data,
     )
-    users = json.loads(response.get_data(as_text=True))
 
-    assert response.status_code == 200
     assert users['data'] == []
 
 
-def test_search_for_users_by_email_handles_incorrect_data_format(notify_db, client):
+def test_search_for_users_by_email_handles_incorrect_data_format(notify_db_session, admin_request):
```

Comment:
This is interesting: I didn't realise we did our own database cleanup. Looks consistent within this file, but I can see a fair few non-`_session` uses out there. I suspect many of them should be changed and we should mark this fixture as semi-private.

How about a quick comment on each fixture to clarify when it should be used?

Reply:
yeah, quick comment makes a lot of sense. i'd love some kind of static analysis tool that tells us how our fixtures are used amongst the various unit tests to identify usages of notify_db

Reply:
i've made a couple of commits to address this (by making `notify_db` private) but they're pretty big so I think I'll put them in a separate PR.